### ISSUE TITLE
[ip] Support Path MTU Discovery

### ISSUE THREAD

**Issue by BrianZill** [demikernel/demikernel#325]
We currently don't support Path MTU Discovery, which is a technique for avoiding in-path packet fragmentation.

See [RFC 1191](https://www.rfc-editor.org/rfc/rfc1191) (or [RFC 8201](https://www.rfc-editor.org/rfc/rfc8201) for the IPv6 version).

**Comment by ppenna** [comment]
@BrianZill I believe that this is a duplicate to https://github.com/demikernel/demikernel/issues/178. 

Please add your extra info there and then close this issue.

**Comment by BrianZill** [comment]
> @BrianZill I believe that this is a duplicate to #178.
> 
> Please add your extra info there and then close this issue.

No, Issue #178 appears to be talking about the local interface MTU, saying that it shouldn't be a hard-coded constant.  Typically, stacks reads an interface MTU from either some configuration information, or directly query it from the NIC (driver) itself.

This issue is about the Path MTU Discovery protocol, which is a means for probing the entire end-to-end path of a particular connection to determine the smallest MTU link on the path.
